### DISCUSSION TITLE
Help understanding Zigbee2MQTT instabilities

### DISCUSSION THREAD

**Discussion by Mneric** [Koenkk/zigbee2mqtt#19678] [General]
Hello,

I am experiencing instabillities with the zigbee2mqtt add-on.

I noticed this after adding 9 new devices, bringing the total number of devices handled by Zigbee2Mqtt to 18.

Sometime, everything is working as exected and then, at some point, devices stop responding.  
Sometime only some devices are irresponsive, sometime all of them are.

Restarting the zigbee2mqtt can solve the issue but it is not always the case. I may need to restart it several time in order to be in a functionnal state again. 
Until the issue appears again...

When in the "broken" state, there are no particular logs outputed bu zigbee2mqtt. When trying to change the state of a device I see a log signaling the change of state but the state never changes.

Here is my setup:
- Home assistant using HassOS image with the following versions:
   - Core 2023.11.2
   - Supervisor 2023.11.0
   - Operating System 11.1
   - user interface 20231030.2
- Zigbee2Mqtt v 1.33.2
- Zigbee coordinator : efr32mg21 based with firmware EZSP v9
- Raspberry Pi 3

I have found [an in-depth guide giving advice to improve the stabillity](https://www.zigbee2mqtt.io/advanced/zigbee/02_improve_network_range_and_stability.html).

However, there are a lot of suggestions and I would like to identify the root cause of my problems before investing in a solution.

Here are some points of attention I noticed:
- "Raspberry Pi 3 may give bad performance"
- efr32mg21  are experimental and not [recommended adapters](https://www.zigbee2mqtt.io/guide/adapters/)
- Maybe I have too many end-devices compared to the number of routers i have.

I already tried multiple orientation of my adapter and is using a extension cable as advised.

Here is the device map of my devices when everything is working fine:
![771912c1ffccad4a110b290e428767fb7d64fc07_2_1035x675](https://github.com/Koenkk/zigbee2mqtt/assets/92851482/bde09cb6-5ae3-421d-a746-933c2dd0062d)

------------------------------------------------------------

In the end, my question is the following : how can I understand what is happening to narrow down the root cause of my issue and try to fix it?

As I said, the logs are not helpfull. However I will copy them next time I am in a broken state.

Thank you!

**Comment by 0verEngineer** [comment]
I find it quite concerning that such discussions remain unanswered....

I don't have enough experience to give you a 100% valid answer but i think your adapter is simply not 100% supported / in beta, look here: https://www.zigbee2mqtt.io/guide/adapters/#recommended
